sim/sample.py

```diff
@@ -4,172 +4,181 @@
+                rate_of_mutations (int):
+                indel_mutation_fraction (int):
+                indel extension probability (int):
+
+            Returns:
+                None  
+        """
+        genome_fasta_path = simulated_genome_path + self.name + '.simulated.simseq.genome.fa'
+
+        output_prefix = simulated_reads_path + self.name
+        
+        run([
+            "dwgsim",
+            "-e", str(self.per_base_error_rate),
+            "-E", str(self.per_base_error_rate),
+            "-i",
+            "-d", str(outer_distance),
+            "-N", str(self.num_read_pairs),
+            "-1", str(read_length),
+            "-2", str(read_length),
+            "-r", str(rate_of_mutations),
+            "-R", str(indel_mutation_fraction),
+            "-X", str(indel_extension_probability),
+            "-y", str(random_dna_probability),
+            "-H",
+            "-z", str(seed),
+            genome_fasta_path,
+            output_prefix
+        ])
+
+        # How dwgsim chooses to name it's output fastq files
+        dwgsim_read_1 = output_prefix + ".bwa.read1.fastq.gz"
+        dwgsim_read_2 = output_prefix + ".bwa.read2.fastq.gz"
+        
+        # Output fastq filenames compatible with btb-seq
+        read_1 = output_prefix + "_S1_R1_X.fastq.gz"
+        read_2 = output_prefix + "_S1_R2_X.fastq.gz"
+        
+        # Rename output fastq files
+        os.rename(dwgsim_read_1, read_1)
+        os.rename(dwgsim_read_2, read_2)
+
 
 class VcfSample(Sample):
     def __init__(self, 
-        predef_snp_path, 
-        seed=1, 
-        per_base_error_rate="0",
-        num_read_pairs = 144997,
-    ):
+                 predef_snp_path, 
+                 seed=1, 
+                 per_base_error_rate="0",
+                 num_read_pairs = 144997,
+                ):
+        
         self.seed = seed
         self.per_base_error_rate = per_base_error_rate
         self.num_read_pairs = math.ceil(num_read_pairs)
-
-        # TODO: check predef_snp_path exists
-        self.predef_snp_path = predef_snp_path
-
-    @property
-    def name(self):
+        
+        if os.path.isfile(predef_snp_path):
+            self.predef_snp_path = predef_snp_path
+        else:
+            raise Exception("Cant find the pre-defined snp path")
+        
         # extract filename
-        #TODO: this will fail if the extension is not 3 charachters long
+        basename = os.path.splitext(os.path.basename(self.predef_snp_path))[0]
         #TODO: this pattern is getting somewhat unweildy.
         #   etiher use something more general or save metadata separately
-        basename = os.path.basename(self.predef_snp_path)[:-4]
-        return f"{type(self).__name__}-{basename}-seed{self.seed}-error{self.per_base_error_rate}-readpairs-{self.num_read_pairs}"
+        self._name = f"{type(self).__name__}-{basename}-seed{self.seed}-error{self.per_base_error_rate}-readpairs-{self.num_read_pairs}"
```

Comment:
I think this is better as a @property. The _name will be wrong if someone changes the value after initialisation